File: python/arrays/replaceElemWithGreatest.py

class Solution:
    def replaceElements(self, arr: list[int],) -> list[int]:
        rightMax = -1

        for i in range(len(arr)-1, -1, -1):
            print(i)
            newMax = max(rightMax, arr[i])
            arr[i] = rightMax
            rightMax = newMax

        return arr

File: javascript/old/arrays_hashing/replaceElemWithGreatest.ts
export const replaceElements = (arr: number[]): number[] => {
  if (arr.length == 0 || arr.length == 1) return [-1];

  return arr.map((item, index) => {
    const max = Math.max(...arr.slice(index + 1, arr.length));
    return isFinite(max) ? max : -1;
  });

  //   const result = [];
  //   for (let i = 0; i < arr.length; i++) {
  //     const max = Math.max(...arr.slice(i + 1, arr.length));

  //     result.push(isFinite(max) ? max : -1);
  //   }

  //   console.log(result);
  //   return result;
};
